src/clr-angular/data/tree-view/models/tree-node.model.spec.ts

```diff
@@ -132,5 +132,119 @@ export default function(): void {
       root.destroy();
       expect(complete).toBeTrue();
     });
+
+    describe('with disabled nodes', function() {
+      it('offers a disabled BehaviorSubject', function() {
+        expect(root.disabled instanceof BehaviorSubject).toBeTrue();
+      });
+
+      it('allows to set the disabled state of a node without propagating', function() {
+        child.setDisabled(true, false, false);
+        expect(child.disabled.value).toBe(true);
+        [root, ...child.children].forEach(n => expect(n.disabled.value).toBe(false));
+      });
+
+      it('emits disabled changes only when it actually changes', function() {
+        let nbChanges = 0;
+        child.disabled.subscribe(_ => nbChanges++);
+        // BehaviorSubject sends us the first state on subscription
+        expect(nbChanges).toBe(1);
+        child.setDisabled(false, false, false);
+        expect(nbChanges).toBe(1);
+        child.setDisabled(true, false, false);
+        expect(nbChanges).toBe(2);
+      });
+
+      it('can propagate the disabled state to parents', function() {
+        // Disabling all children of a node will make it disabled
+        child.children.forEach(c => c.setDisabled(true, true, false));
+        expect(child.disabled.value).toBe(true);
+        // Once the parent is disabled, a child cannot be enabled
+        child.children[0].setDisabled(false, false, false);
+        expect(child.children[1].disabled.value).toBe(true);
+      });
+
+      it('can propagate the disabled state to children', function() {
+        // Disabling a parent disables all the children down the tree
+        root.setDisabled(true, false, true);
+        [...root.children, ...child.children].forEach(n => expect(n.disabled.value).toBe(true));
```

Comment:
I think it would be more efficient to add up children that are true and compare that against the total length of items, so we only run one `expect` check instead of N `expect`s.